client/src/components/useraccess/Register.jsx

```diff
@@ -0,0 +1,100 @@
+import React, {PropTypes} from 'react';
+import { Link, Redirect } from 'react-router-dom';
+import { connect } from 'react-redux';
+
+import Navbar from '../navbar/Navbar';
+import RegistrationForm from './Forms/RegistrationForm';
+import Background from '../Background/Background';
+import * as UserAcessActions from '../../Actions/userAccessAction';
+
+
+
+/**
+ * @class Register
+ * @classdesc returns the component for user signup
+ */
+class Register extends React.Component {
+  constructor(props, context) {
+    super(props, context);
+    this.state = {
+      userData: Object.assign({}, this.props.initialUserData),
+      redirect: false,
+      error: ''
+    }
+
+    this.handleUserInput = this.handleUserInput.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+
+  //  Handle user input
+  handleUserInput(event) {
+    // get name of the target element
+      const name = event.target.name; 
+
+    // copy key in userData and assign to supllied data
+      let suppliedData = Object.assign({}, this.state.userData);
+
+    // set each key of the supllied value
+      suppliedData[name] = event.target.value
+    
+    // update userData state to the supllied data 
+      return this.setState({userData: suppliedData})
```

Comment:
I think you have over-engineered this, I'm guessing you want to edit just a single property inside `this.state.userData`. You can think of a better way of refactoring it